src/screenComponents/databaseView.cpp

```diff
@@ -69,35 +61,90 @@ void DatabaseViewComponent::fillListBox()
 {
     item_list->setOptions({});
     item_list->setSelectionIndex(-1);
-    if (!selected_entry)
+
+    // indices of child or sibling pages in the science_databases vector
+    std::vector<unsigned> children_idx;
+    std::vector<unsigned> siblings_idx;
+    P<ScienceDatabase> parent_entry;
+
+    for (unsigned idx=0; idx<ScienceDatabase::science_databases.size(); idx++)
     {
-        foreach(ScienceDatabase, sd, ScienceDatabase::science_databases)
+        P<ScienceDatabase> sd = ScienceDatabase::science_databases[idx];
+        if (!sd) continue;
+
+        if(selected_entry)
         {
-            item_list->addEntry(sd->getName(), sd->getName());
+            if(sd->getId() == selected_entry->getParentId())
+            {
+                parent_entry = sd;
+            }
+            if(sd->getParentId() == selected_entry->getParentId())
+            {
+                siblings_idx.push_back(idx);
+            }
+            if(sd->getParentId() == selected_entry->getId())
+            {
+                children_idx.push_back(idx);
+            }
+        }
+        else
+        {
+            if(sd->getParentId() == 0)
+            {
+                siblings_idx.push_back(idx);
+            }
+        }
+    }
+
+    if(selected_entry)
+    {
+        if (children_idx.size() != 0)
+        {
+            item_list->addEntry(tr("Back"), std::to_string(selected_entry->getParentId()));
         }
-    }else{
-        item_list->addEntry(tr("Back"), "");
-        foreach(ScienceDatabase, sd, selected_entry->items)
+        else if(parent_entry)
         {
-            item_list->addEntry(sd->getName(), sd->getName());
+            item_list->addEntry(tr("Back"), std::to_string(parent_entry->getParentId()));
+        }
+    }
+    if(children_idx.size() > 0)
```

Comment:
I think this code becomes a bit smaller and simpler if we assign `children_idx` to `siblings_idx` if `children_idx` is not empty, and then always use the 2nd path.